SICP/2-1/ex-2-3.md

```diff
@@ -0,0 +1,12 @@
+# SICP 2-3
+## 문제 설명
+- `rectangle`, `area`, `perimeter` 프로시저를 설계하는 것이 목표.
+- `rectangle`의 구현은 두가지 이상이 될 수 있음.
+- `area`, `perimeter`는 `rectangle`의 구현과 상관없이 동일한 방법으로 사용할 수 있어야 함. 
+
+## 풀이 계획
+> 1. area, perimeter 프로시저 정의  
+>   - rectangle의 selector, constructor 결정
+> 2. rectangle 프로시저 구현
+> 3. rectangle 프로시저 변경
+>   - 동일한 area, perimeter를 출력하는지 확인
```

Comment:
이것 안함.

Reply:
점 네개로 데이터 표현을 바꿔서 수행. 
넓이를 구하는 방식이 바뀌지 않는 것을 확인. 

원문. 
> Can you design your system with suitable abstraction barriers,
so that the same perimeter and area procedures will work with either representation?